src/d3a/models/area/stats.py

```diff
@@ -183,18 +184,22 @@ def get_market_stats(self, market_slot_list, dso=False):
 
         return out_dict
 
-    def aggregate_exported_imported_energy(self, area):
-        if self.current_market is None:
+    def _aggregate_exported_imported_energy(self):
+        if self._area.current_market is None:
             return None
 
-        child_names = [area_name_from_area_or_iaa_name(c.name) for c in area.children]
-        for trade in self.current_market.trades:
-            if child_buys_from_area(trade, area.name, child_names):
-                add_or_create_key(self.exported_energy, self.current_market.time_slot,
-                                  trade.offer.energy)
-            if area_sells_to_child(trade, area.name, child_names):
-                add_or_create_key(self.imported_energy, self.current_market.time_slot,
-                                  trade.offer.energy)
+        self.imported_energy = {}
+        self.exported_energy = {}
+
+        child_names = [area_name_from_area_or_iaa_name(c.name) for c in self._area.children]
+        if getattr(self.current_market, 'trades', None) is not None:
+            for trade in self.current_market.trades:
+                if child_buys_from_area(trade, self._area.name, child_names):
+                    add_or_create_key(self.exported_energy, self.current_market.time_slot,
+                                      trade.offer.energy)
+                if area_sells_to_child(trade, self._area.name, child_names):
+                    add_or_create_key(self.imported_energy, self.current_market.time_slot,
+                                      trade.offer.energy)
```

Comment:
I think there is no member self.current_market here. You probably meant self.area.current_market.

Reply:
There is `src/d3a/models/area/stats.py:166`

Reply:
aahh, but then you dont need to parse the area object (you can get rid of self._area). Perfect!

Reply:
I cant get rid of `self._area` as I need it to accumulate `child_names `